projects/kit/components/pdf-viewer/pdf-viewer.component.ts

```diff
@@ -1,33 +1,103 @@
 import {AnimationOptions} from '@angular/animations';
-import {ChangeDetectionStrategy, Component, HostBinding, Inject} from '@angular/core';
-import {TuiDialog} from '@taiga-ui/cdk';
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    HostBinding,
+    Inject,
+} from '@angular/core';
+import {WINDOW} from '@ng-web-apis/common';
+import {isFirefox, isSafari, TuiDestroyService, TuiDialog} from '@taiga-ui/cdk';
 import {
     TUI_ANIMATION_OPTIONS,
     TUI_CLOSE_WORD,
     tuiFadeIn,
     tuiSlideInTop,
 } from '@taiga-ui/core';
 import {POLYMORPHEUS_CONTEXT} from '@tinkoff/ng-polymorpheus';
-import {Observable} from 'rxjs';
+import {fromEvent, Observable, race, timer} from 'rxjs';
+import {filter, map, takeUntil} from 'rxjs/operators';
 
 import {TuiPdfViewerOptions} from './pdf-viewer-options';
 
+// @dynamic
 @Component({
     selector: 'tui-pdf-viewer',
     templateUrl: './pdf-viewer.template.html',
     styleUrls: ['./pdf-viewer.style.less'],
     changeDetection: ChangeDetectionStrategy.OnPush,
     animations: [tuiSlideInTop, tuiFadeIn],
+    providers: [TuiDestroyService],
 })
-export class TuiPdfViewerComponent<I, O> {
+export class TuiPdfViewerComponent<I, O> implements AfterViewInit {
     @HostBinding('@tuiSlideInTop')
     @HostBinding('@tuiFadeIn')
     readonly animation = {value: '', ...this.options} as const;
 
+    emptyPreview = false;
+
     constructor(
         @Inject(TUI_ANIMATION_OPTIONS) private readonly options: AnimationOptions,
         @Inject(TUI_CLOSE_WORD) readonly closeWord$: Observable<string>,
         @Inject(POLYMORPHEUS_CONTEXT)
         readonly context: TuiDialog<TuiPdfViewerOptions<I>, O>,
+        @Inject(ElementRef)
+        private readonly elementRef: ElementRef<Element>,
+        @Inject(ChangeDetectorRef)
+        private readonly cd: ChangeDetectorRef,
+        @Inject(WINDOW)
+        private readonly windowRef: Window & {chrome: unknown},
+        @Inject(TuiDestroyService)
+        private readonly destroy$: TuiDestroyService,
     ) {}
+
+    ngAfterViewInit(): void {
+        const iframe: HTMLIFrameElement | null =
+            this.elementRef.nativeElement.querySelector('iframe');
+
+        if (iframe) {
+            race(timer(1000), fromEvent(iframe, 'load'))
+                .pipe(
+                    map(() => this.hasPdfInsideIframe(iframe)),
+                    filter(pdfVisible => !pdfVisible),
+                    takeUntil(this.destroy$),
+                )
+                .subscribe(() => {
+                    // hide default preview (chrome/safari) for download pdf
+                    iframe.style.setProperty('display', 'none');
+                    this.emptyPreview = true;
+                    this.cd.detectChanges();
```

Comment:
We can deal with it without extra explicit subscription and extra selector.
Use `*tuiLet` inside template to subscribe to this observable.
Then in template:
```html
<header class="t-header" *tuiLet="iframe$ | async as iframeAvailable">
    ...
    <section
        polymorpheus-outlet
        class="t-content"
        [content]="iframeAvailable ? context.content : errorMessageRef"
        <iframe
            class="t-iframe"
            [class.t-iframe_hidden]="!iframeAvailable"
        ></iframe>
    </section>
    
    <ng-template #errorMessageRef>...</ng-template>
</header>

```